src/Record.js

```diff
@@ -105,10 +105,7 @@ export class Record {
 
   equals(other) {
     return (
-      this === other ||
-      (other &&
-        this._keys === other._keys &&
```

Comment:
I removed this `_keys_` comparison as the `recordSeq` does contains the keys and values